internal/build/image_builder.go

```diff
@@ -55,14 +55,14 @@ func DefaultImageBuilder(b *dockerImageBuilder) ImageBuilder {
 }
 
 func DefaultConsole() console.Console {
-	out := os.Stdout
+	out := output.File()
```

Comment:
i don't think this will work like you expect? Notice that DefaultConsole is called before `output.CaptureAll`, so the output will not get captured here.

As an aside, this PR is a pretty good example of the order-of-initialization bugs that come up when you use global state, and which dependency injection tries to resolve!

I would probably change this PR by
1) Remove the global state in output.CaptureAll
2) Make output.CaptureAll return the new output writer, so that you can use it as a provider
3) Add output.CaptureAll to your DI graph (instead of DefaultOut), to ensure it gets initialized first

Reply:
That's a good idea, I'll do that!